src/field.rs

```diff
@@ -0,0 +1,46 @@
+use core::{
+    fmt::Debug,
+    iter::{Product, Sum},
+    ops::{Add, AddAssign, Mul, MulAssign, Neg, Sub, SubAssign},
+};
+
+pub trait Field:
+    Sized
+    + Eq
+    + Copy
+    + Clone
+    + Default
+    + Neg<Output = Self>
+    + From<u64>
+    + Add<Output = Self>
+    + Sub<Output = Self>
+    + Mul<Output = Self>
+    + Sum
+    + Product
+    + for<'a> Add<&'a Self, Output = Self>
+    + for<'a> Sub<&'a Self, Output = Self>
+    + for<'a> Mul<&'a Self, Output = Self>
+    + for<'a> Sum<&'a Self>
+    + for<'a> Product<&'a Self>
+    + AddAssign
+    + SubAssign
+    + MulAssign
+    + for<'a> AddAssign<&'a Self>
+    + for<'a> SubAssign<&'a Self>
+    + for<'a> MulAssign<&'a Self>
+    + Send
+    + Sync
+    + Debug
+    + 'static
```

Comment:
Compared to `halo2proofs::arithmetic::Field`:

Removed:
    + ConditionallySelectable
    + ConstantTimeEq

Any specific reasons?

Reply:
We don't use these traits in our code.